src/service/resources-impl.js

```diff
@@ -1272,6 +1313,23 @@ export class Resources {
       }
     }
   }
+
+  /**
+   * Removes the task resource from the discovered first viewport
+   * resources array.
+   * @param {!TaskDef} task
+   */
+  checkForFirstViewportElementsWhenTaskDone_(task) {
```

Comment:
We can certainly hook into `layoutComplete_` of Resource class. We will then need some sort of `whenFirstLayoutComplete` promise there.

For now, let's just wait for DOMReady in the performance class and call getResourcesInViewport that will simply return an array of resources.

`prerenderAllowed` is already exposed as public method on Resource class.